library/pk.c

```diff
@@ -464,16 +464,12 @@ int mbedtls_pk_get_psa_attributes(const mbedtls_pk_context *pk,
             int sign_ok = (pk_type != MBEDTLS_PK_ECKEY_DH);
             int derive_ok = (pk_type != MBEDTLS_PK_ECDSA);
 #if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
-            psa_key_attributes_t old_attributes = PSA_KEY_ATTRIBUTES_INIT;
-            psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
-            status = psa_get_key_attributes(pk->priv_id, &old_attributes);
-            if (status != PSA_SUCCESS) {
-                return MBEDTLS_ERR_PK_BAD_INPUT_DATA;
+            psa_ecc_family_t family = pk->ec_family;
+            size_t bits = pk->ec_bits;
+            int has_private = 0;
```

Comment:
Minor: I'd prefer
```
int has_private = pk->priv_id != MBEDTLS_SVC_KEY_ID_INIT;
```
but that might be a matter of taste, and definitely no worth another round if there's nothing else.